marlowe-playground-client/src/Examples/JS/Contracts.purs

```diff
@@ -96,6 +96,111 @@ escrow =
     return contract;
 """
 
+escrowWithCollateral :: String
+escrowWithCollateral =
+  """
+ import {
+    PK, Role, Account, Party, ada, AvailableMoney, Constant, NegValue, AddValue,
+    SubValue, MulValue, Scale, ChoiceValue, SlotIntervalStart, SlotIntervalEnd,
+    UseValue, Cond, AndObs, OrObs, NotObs, ChoseSomething, ValueGE, ValueGT,
+    ValueLT, ValueLE, ValueEQ, TrueObs, FalseObs, Deposit, Choice, Notify,
+    Close, Pay, If, When, Let, Assert, SomeNumber, AccountId, ChoiceId, Token,
+    ValueId, Value, EValue, Observation, Bound, Action, Payee, Case, Contract
+} from 'marlowe-js';
+
+(function (): Contract {
+
+    /* Parties */
+    const alice : Party = Role("alice");
+    const bob : Party = Role("bob");
+    const blackhole : Party = Role("blackhole");
+
+    /* Value under escrow */
+    const price : SomeNumber = 450n;
+    const aliceCollateral : SomeNumber = 4500n;
+    const bobCollateral : SomeNumber = 4500n;
+
+    /* helper function to build Actions */
+
+    const choiceName : string = "choice";
+
+    const choiceIdBy = function (party : Party) : ChoiceId {
+                        return ChoiceId(choiceName, party);
+                    }
+
+    const choiceBy = function(party : Party, bounds : [Bound]) : Action {
+                        return Choice(choiceIdBy(party), bounds);
+                    };
+
+
+    const choiceValueBy = function(party : Party) : Value {
+                            return ChoiceValue(choiceIdBy(party));
+                        };
+
+    /* Names for choices */
+
+    const both : [Bound]   = [Bound(0n, 1n)];
+
+    /* Name choices according to person making choice and choice made */
+
+
+    const aliceChoice : Action = choiceBy(alice, both);
+
+    const bobChoice : Action = choiceBy(bob, both);
+
+    /* the values chosen in choices */
+
+    const aliceChosen : Value = choiceValueBy(alice);
+    const bobChosen : Value = choiceValueBy(bob);
+
+    /* The contract to follow when Alice and Bob disagree, or if
+    Carol has to intervene after a single choice from Alice or Bob. */
+
+    const destroyCollaterals : Contract = Pay(alice, Party(blackhole), ada, aliceCollateral, Pay(bob, Party(blackhole), ada, bobCollateral, Close));
+
+    /* The contract to follow when Alice and Bob have made the same choice. */
+
+    const agreement : Contract = If(ValueEQ(aliceChosen, 0n),
+                                    Pay(alice, Party(bob), ada, price, Close),
+                                    Close);
+
+    /* Inner part of contract */
+
+    const inner : Contract = When([Case(aliceChoice,
+                            When([Case(bobChoice,
+                                        If(ValueEQ(aliceChosen, bobChosen),
+                                            agreement,
+                                            destroyCollaterals))],
+                                    60n, destroyCollaterals))],
+                            40n, destroyCollaterals);
```

Comment:
Two observations, maybe they are intentional, but they are a bit odd for me:
- If there is no agreement, Alice will still get the deposit back, even though the collateral is destroyed.
- Alice can force Bob into losing the collateral by not answering (Bob can do the same, so that is consistent), but I wonder, if Alice doesn't answer on time, would make more sense to assume Bob is right, and viceversa?

Reply:
1) it is intentional, because idk who's right who's wrong - so I chose to believe that nothing happened by default (choosing to believe the opposite is totally equivalent so i don't think there is right answer :) ) . Parties could adjust alice's collateral to account for it.

2) Then "not answering" would mean agreement (so cat was transferred). Idk is it right or wrong - my idea here was to "punish both parties for absense of consensus", 

we could say "being quiet is a sign of agreement" - I think it's okay, but really don't see difference. Besides, idk how realistic it is but "incentive to make the other party quiet" sounds scarry to me.